server/src/test/java/org/opensearch/cluster/coordination/PublicationTransportHandlerTests.java

```diff
@@ -111,10 +133,182 @@ public void writeTo(StreamOutput out) throws IOException {
 
         OpenSearchException e = expectThrows(
             OpenSearchException.class,
-            () -> handler.newPublicationContext(new ClusterChangedEvent("test", unserializableClusterState, clusterState))
+            () -> handler.newPublicationContext(new ClusterChangedEvent("test", unserializableClusterState, clusterState), false, null)
         );
         assertNotNull(e.getCause());
         assertThat(e.getCause(), instanceOf(IOException.class));
         assertThat(e.getCause().getMessage(), containsString("Simulated failure of diff serialization"));
     }
+
+    public void testHandleIncomingPublishRequestWhenNoCurrentPublishRequest() {
```

Comment:
rename the test to testHandleIncomingRemotePublishRequestWhenPublishingToSelfFailed

Reply:
Also for all the tests add remote in the test name